src/Compilers/CSharp/Portable/Syntax/AnonymousFunctionExpressionSyntax.cs

```diff
@@ -0,0 +1,18 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+namespace Microsoft.CodeAnalysis.CSharp.Syntax
+{
+    public partial class AnonymousFunctionExpressionSyntax
+    {
+        /// <summary>
+        /// Either the <see cref="Block"/> if it is not <c>null</c> or the
+        /// <see cref="ExpressionBody"/> otherwise.
+        /// </summary>
+        public CSharpSyntaxNode Body => Block ?? (CSharpSyntaxNode)ExpressionBody;
+
+        public AnonymousFunctionExpressionSyntax WithBody(CSharpSyntaxNode body)
+            => body is BlockSyntax block
+                ? WithBlock(block)
+                : WithExpressionBody((ExpressionSyntax)body);
```

Comment:
this will be a common pattern.  We introduce the binary compat bridge methods that just take an arbitrary syntax node.  And we then defer to either hte block or expression form accordingly.